webapp/src/shared/components/Status/Status.tsx

```diff
@@ -0,0 +1,55 @@
+import './Status.css';
+
+type NoStatusProps = {
+  offline?: never;
+  online?: never;
+  playing?: never;
+};
+
+type OfflineStatusProps = {
+  offline: boolean;
+  online?: never;
+  playing?: never;
+};
+
+type OnlineStatusProps = {
+  offline?: never;
+  online: boolean;
+  playing?: never;
+};
+
+type PlayingStatusProps = {
+  offline?: never;
+  online?: never;
+  playing: boolean;
+};
+
+type StatusProps =
+  | NoStatusProps
+  | OfflineStatusProps
+  | OnlineStatusProps
+  | PlayingStatusProps;
+
+const createVariant = (variant: { [key: string]: boolean }): string => {
+  for (const [key, value] of Object.entries(variant)) {
+    if (value) {
+      return key;
+    }
+  }
+  return 'offline';
+};
+
+export default function Status({
+  online = false,
+  playing = false,
+  offline = false,
+}: StatusProps) {
+  const variant = createVariant({ offline, online, playing });
+
+  return (
+    <div className="status">
+      <div className={`status-icon status-icon-${variant}`} />
+      <div className={`text-color-${variant} caption-regular`}>{variant}</div>
+    </div>
+  );
+}
```

Comment:
I think you may over engineer the solution because this could be easily solved in this case with string literal types. This would imply changing how we pass the prop to the component:

```tsx
      <Status variant='online' />
```

I know you've came up to this solution because in the issue we (I :sweat: ) described to have boolean props. But after seeing your implementation I think string literal types are a really better fit here. What do you think?

This being said, whatever you decide I'll agree with it!

```suggestion
type StatusType = 'online' | 'offline' | 'playing';

type StatusProps = {
    variant: StatusType
}

export default function Status({
    variant
}: StatusProps) {

  return (
    <div className="status">
      <div className={`status-icon status-icon-${variant}`} />
      <div className={`text-color-${variant} caption-regular`}>{variant}</div>
    </div>
  );
}
```

Reply:
You are right, and this is a lot easier to read :100:. Thank you.